src/Client/ClientBase.cpp

```diff
@@ -938,6 +938,21 @@ void ClientBase::initKeystrokeInterceptor()
     }
 }
 
+
+String ClientBase::appendSmileyIfNeeded(const String & prompt_)
+{
+    static constexpr String smiley = ":) ";
+
+    if (prompt_.empty())
+        return smiley;
+
+    if (prompt_.ends_with(smiley))
+        return prompt_;
+
+    return prompt_ + smiley;
```

Comment:
It should be `return prompt_ + " " + smiley;` so that `--prompt test` leads to a prompt like `test :) ` (with a space between the prompt and the smiley).

The reason I only checked for `:) ` at the end of the prompt is so that it's possible to specify a minimalist prompt with e.g. `<prompt>:) </prompt>` if one wants to maximize available space.

Reply:
You're right, fixed, thanks.